### PR TITLE
Jesse/issue 3

### PR DESCRIPTION
**Pull request by vc-ca**
This issue is incomplete by the 5/25 deadline, but I'll finish it by 5/27 if that's an option.

I was able to fix:
1. Messages not properly sending. - Required an async function.
 2. Messages not displaying in the correct order - data retrieved from DB just needed to be output in reverse.
 
 I was unable to complete the security issue "where a user can send messages to any conversation" bug.

### CODE REVIEW

**Comment by vc-ca** [comment]
All issues are now resolved with most recent commit

File: server/db/db.js [hunk 1]
```diff
@@ -1,7 +1,10 @@
 const Sequelize = require("sequelize");
+require("dotenv").config();
```
Comment (johnsciutto) [inline server/db/db.js:2]: Isn't this already running in the **server/bin/www** file?

**Review by johnsciutto — COMMENTED** [review]
Hi @vc-ca,
Nice work overall 👍🏻 
I left a few small comments 🙂

File: server/db/db.js (hunk 1 above)
Comment (vc-ca) [reply server/db/db.js:2]: Not sure why I added that one

File: server/routes/api/conversations.js [hunk 2]
```diff
@@ -20,7 +20,7 @@ router.get("/", async (req, res, next) => {
         },
       },
       attributes: ["id"],
-      order: [[Message, "createdAt", "DESC"]],
+      order: [[Message, "createdAt", "ASC"]],
```
[the remark this answers is not in the document]
Comment (vc-ca) [reply server/routes/api/conversations.js]: I don't think I even noticed this! I believe I've now found the issue and resolved it

**Comment by vc-ca** [comment]
I believe I have now resolved the message order problem.

File: client/src/store/utils/reducerFunctions.js [hunk 3]
```diff
@@ -14,7 +15,7 @@ export const addMessageToStore = (state, payload) => {
   return state.map((convo) => {
     if (convo.id === message.conversationId) {
       const convoCopy = { ...convo };
-      convoCopy.messages.push(message);
+      convoCopy.messages.unshift(message);
```
Comment (johnsciutto) [inline client/src/store/utils/reducerFunctions.js:18]: This works. However, it's quite inefficient (the more messages you add to the conversation, the more time this operation takes) 🤔

Comment (vc-ca) [reply client/src/store/utils/reducerFunctions.js]: Interesting. I was not aware that unshift was inefficient. I've replaced this with a spread operator.

**Review by bonnieli — APPROVED** [review]

**Pull request merged**